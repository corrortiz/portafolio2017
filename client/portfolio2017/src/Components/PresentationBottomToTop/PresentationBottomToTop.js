import React, { Component } from 'react';
import PropTypes from 'prop-types';
//HOC for global script
import GlobalsConnect from '../../HOC/GlobalsConnect/GlobalsConnect';
//Locale
import { lenguajeSelector } from '../../Store/Actions/globals';
import { Meta, MetaText } from '../../Assets/diccionary';
//MUI Component
import Grid from 'material-ui/Grid';
import { withStyles } from 'material-ui/styles';
//Internal Component
import Logo from '../../images/SVG/Logo';
//CSS in JS Styles
const styles = theme => ({
  root: {}
});

/**
 * This class represents a common section of two rows
 * one with information and the other with image botton right to top
 */
export class PresentationBottomToTop extends Component {
  render() {
    const { lenguaje } = this.props.globals;
    const { classes } = this.props;

    return (
      <div className={classes.root}>
        <Grid container spacing={8} className="PresentationBottomToTop">
          <Grid item xs={12} className="PresentationBottomToTop__one">
            <Logo />
          </Grid>
          <Grid item xs={12} className="PresentationBottomToTop__two">
            <h1 className="PresentationBottomToTop__two__title">
              {lenguajeSelector(lenguaje, Meta)}
            </h1>
            <p className="PresentationBottomToTop__two__text">
              {lenguajeSelector(lenguaje, MetaText)}
            </p>
          </Grid>
        </Grid>
      </div>
    );
  }
}

PresentationBottomToTop.propTypes = {
  globals: PropTypes.shape({
    lenguaje: PropTypes.string.isRequired
  })
};

PresentationBottomToTop = withStyles(styles)(PresentationBottomToTop);

export default GlobalsConnect(PresentationBottomToTop);
