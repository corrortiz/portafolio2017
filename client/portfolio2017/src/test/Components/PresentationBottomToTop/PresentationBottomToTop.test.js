import React from 'react';
import { shallow } from 'enzyme';
import { PresentationBottomToTop } from '../../../Components/PresentationBottomToTop/PresentationBottomToTop';

describe('PresentationBottomToTop Component', () => {
  it('renders without crashing', () => {
    shallow(<PresentationBottomToTop />);
  });

  it('should compare a snapshops of PresentationBottomToTop', () => {
    const wrapper = shallow(<PresentationBottomToTop />);
    expect(wrapper).toMatchSnapshot();
  });
});
