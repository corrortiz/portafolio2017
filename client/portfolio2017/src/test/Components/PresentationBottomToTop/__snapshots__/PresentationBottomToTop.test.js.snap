// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PresentationBottomToTop Component should compare a snapshops of PresentationBottomToTop 1`] = `
<PresentationBottomToTop
  classes={
    Object {
      "root": "PresentationBottomToTop-root-1",
    }
  }
/>
`;
